app/controllers/purchases_controller.rb

```diff
@@ -0,0 +1,60 @@
+class PurchasesController < ApplicationController
+  def index
+    @purchases_nav = true
+    @category = Category.includes(:purchases).find(params[:category_id])
+    @purchases = @category.purchases.order(created_at: :desc)
+  end
+
+  def show
+    @each_purchase_nav = true
+    @category = Category.includes(:purchases).find(params[:category_id])
+    @purchase = @category.purchases.includes(:categories).find(params[:id])
+  end
+
+  def new
+    @new_purchase_nav = true
+    @category = Category.find(params[:category_id])
+    @purchase = purchase.new
```

Comment:
- [ ] Please fix this name error as it's not working.You are calling a class method and it should start with a capital letter.